addons/account/models/account_invoice.py

```diff
@@ -1867,7 +1867,8 @@ def _get_price_tax(self):
         ondelete='cascade', index=True)
     invoice_type = fields.Selection(related='invoice_id.type', readonly=True)
     uom_id = fields.Many2one('uom.uom', string='Unit of Measure',
-        ondelete='set null', index=True, oldname='uos_id')
+        ondelete='set null', index=True, oldname='uos_id', domain="[('category_id', '=', product_uom_category_id)]")
+    product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
```

Comment:
`readonly=True` for all the category related fields in this PR to avoid customers changing the category of a uom by accident (even though it's invisible in the view, they might use studio or smthg else)